test/functional/p2p_handshake.py

```diff
@@ -88,6 +89,12 @@ def run_test(self):
         with node.assert_debug_log([f"feeler connection completed"]):
             self.add_outbound_connection(node, "feeler", NODE_NONE, wait_for_disconnect=True)
 
+        self.log.info("Check that connecting to ourself leads to immediate disconnect")
+        with node.assert_debug_log(["connected to self", "disconnecting"]):
+            node_listen_addr = f"127.0.0.1:{p2p_port(0)}"
+            node.addconnection(node_listen_addr, "outbound-full-relay", self.options.v2transport)
```

Comment:
I like how this is using `self.options.v2transport` as the argument, enabling the script list in `test_runner` to specify whether or not v2transport should be used for the test.